background/main.js

```diff
@@ -59,33 +77,30 @@ torrentToWeb.createAdapter = function (callback) {
     });
 };
 
-torrentToWeb.determineFilename = function (response) {
-    let filename = null;
-    let result;
-    let filenameHeader = response.headers.get('filename');
-    let nameHeader = response.headers.get('name');
-    let contentHeader = response.headers.get('content-disposition');
-
-    if (filenameHeader !== null && filenameHeader !== '') {
-        console.log('Found filename in "filename" header');
-        filename = filenameHeader;
-    } else if (nameHeader !== null && nameHeader !== '') {
-        console.log('Found filename in "name" header');
-        filename = nameHeader;
-    } else if (contentHeader !== null && (result = contentHeader.match(/filename=(?:"([^"]+)"|([^;]+);?)/))) {
-        console.log('Found filename in "content-disposition" header');
-        filename = (typeof result[1] !== 'undefined' ? result[1] : result[2]);
-    } else {
-        console.log('Falling back to filename from URL');
-        filename = new URL(response.url).pathname.split('/').pop();
-    }
-
-    filename = filename.replace(/[^0-9a-zA-Z.]+/g, '.');
-    filename = filename.replace(/\.{2,}/g, '.');
-    filename = filename.replace(/(^\.+|\.+$)/g, '');
-
-    console.log('Using filename "' + filename + '"');
-    return filename;
+torrentToWeb.determineFilename = function (blob) {
+    return new Promise((resolve) => {
+        blob.arrayBuffer().then((buffer) => {
+            let torrent;
+            try {
+                torrent = decode(buffer);
+            } catch (error) {
+                resolve(false);
+                return;
+            }
+
+            if (torrent.info && (torrent.info.name || torrent.info.name.utf8)) {
+                if (torrent.info.name.utf8) {
+                    resolve(torrent.info.name.utf8);
+                    return;
+                }
+
+                resolve(torrent.info.name);
```

Comment:
According to the specs, this is the name of the file or directory, of which the torrent was created. Thus shouldn't we append `.torrent` somewhere?

Reply:
No. Again: The name of the torrent file is actually completely irrelevant. The client does not take this name anyway. It just works like with executables on Unix compared to Windows: On windows, it must be named .exe but on Unix, the name could be anything as long as the ELF-Header inside the file is valid.

Reply:
Actually, a competing product always uses constant "temp.torrent" for the name during upload and still in the clients, display the name that is stored in info.name :-)

Reply:
And: Of course, I tested that against all currently supported clients.